src/arctica-greeter.vala

```diff
@@ -286,6 +301,16 @@ public class ArcticaGreeter : Object
         /* Make sure the given session actually exists. Return it if it does.
          * otherwise, return the default session.
          */
+        var excluded_sessions = AGSettings.get_strv (AGSettings.KEY_EXCLUDED_SESSIONS);
+        var includeonly_sessions = AGSettings.get_strv (AGSettings.KEY_INCLUDEONLY_SESSIONS);
+
+        if ((includeonly_sessions.length > 0) && (!(session in includeonly_sessions))) {
```

Comment:
Same here, adapted to the different action:

```vala
        if (includeonly_sessions.length > 0) {
            if (!(session in includeonly_sessions)) {
                session = null;
            }
        }
        else if (session in excluded_sessions) {
            session = null;
        }
```

Reply:
dito, please adjust